metrics/whisper/tracer_prometheus.go

```diff
@@ -0,0 +1,49 @@
+// +build metrics,prometheus
+
+// Package whisper collects Whisper envelope metrics using Prometheus.
+package whisper
+
+import (
+	"strconv"
+
+	whisper "github.com/ethereum/go-ethereum/whisper/whisperv5"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	envelopeCounter = prometheus.NewCounterVec(
```

Comment:
unfortunate that there is no analog for map in prometheus library, but what about envelope_new_counter?